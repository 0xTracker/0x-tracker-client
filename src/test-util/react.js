import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
