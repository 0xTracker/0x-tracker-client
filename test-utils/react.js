import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
