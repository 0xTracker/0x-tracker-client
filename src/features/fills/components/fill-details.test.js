import _ from 'lodash';
import React from 'react';

import { renderWithRouter } from '../../../test-util/react';
import FillDetails from './fill-details';

const simpleFill = {
  assets: [
    {
      amount: '0.96955',
      price: { USD: 190.08 },
      tokenAddress: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
      tokenSymbol: 'WETH',
      tokenType: 'Wrapped Ether',
      traderType: 'maker',
      type: 'erc-20',
    },
    {
      amount: '5000',
      price: { USD: 0.036858412800000004 },
      tokenAddress: '0x4946fcea7c692606e8908002e55a582af44ac121',
      tokenSymbol: 'FOAM',
      tokenType: 'FOAM Token',
      traderType: 'taker',
      type: 'erc-20',
    },
  ],
  date: '2019-10-10T09:13:48.000Z',
  feeRecipient: '0xa258b39954cef5cb142fd567a46cddb31a670124',
  id: '5d9ef747e3fcbb000467cbf5',
  makerAddress: '0x2cfb4faaece38d5d03ef1ec9dfdf34964d6ab1b3',
  makerFee: { USD: 0, ZRX: '0' },
  orderHash:
    '0x8104d4c4a32da7f1df357779fe6b8fc11dc6f4287de54566fe22eb35a74b0c71',
  protocolVersion: 2,
  relayer: {
    imageUrl: 'https://0xtracker.com/assets/logos/radar-relay.png',
    name: 'Radar Relay',
    slug: 'radar-relay',
  },
  senderAddress: '0x0dc411b17d337af85d83ea5a3577d09132aae866',
  status: 'successful',
  takerAddress: '0x0dc411b17d337af85d83ea5a3577d09132aae866',
  takerFee: { USD: 0, ZRX: '0' },
  totalFees: { USD: 0, ZRX: '0' },
  transactionHash:
    '0x68e018d2c97cf76af6bae65297e6f5b6bc9680b95b828bdea5eb03a0b36ebc77',
  value: { USD: 184.292064 },
};

const simpleScreenSize = { lessThan: { sm: false } };

beforeAll(() => {
  // eslint-disable-next-line no-extend-native
  Date.prototype.getTimezoneOffset = _.constant(180); // Mock timezone as UTC-3
});

describe('fillDetails component', () => {
  it('should render V2 fill', () => {
    const { asFragment } = renderWithRouter(
      <FillDetails fill={simpleFill} screenSize={simpleScreenSize} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render V3 fill', () => {
    const fill = {
      ...simpleFill,
      makerFee: undefined,
      protocolFee: {
        ETH: '0.0001',
        USD: 0.2,
      },
      takerFee: undefined,
    };

    const { asFragment } = renderWithRouter(
      <FillDetails fill={fill} screenSize={simpleScreenSize} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
