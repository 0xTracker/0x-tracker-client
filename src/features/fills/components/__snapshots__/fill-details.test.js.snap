// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fillDetails component should render V2 fill 1`] = `
<DocumentFragment>
  <dl
    class="fill-details__FillDetailList-sc-5nb7jb-0 mycsW"
  >
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Transaction Hash:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__OutboundLink-sc-1m1n17q-0 dFkevI"
        href="https://etherscan.io/tx/0x68e018d2c97cf76af6bae65297e6f5b6bc9680b95b828bdea5eb03a0b36ebc77"
        rel="noopener noreferrer"
        target="_blank"
      >
        0x68e018d2c97cf76af6bae65297e6f5b6bc9680b95b828bdea5eb03a0b36ebc77
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Order Hash:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/search?q=0x8104d4c4a32da7f1df357779fe6b8fc11dc6f4287de54566fe22eb35a74b0c71"
      >
        0x8104d4c4a32da7f1df357779fe6b8fc11dc6f4287de54566fe22eb35a74b0c71
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Sender Address:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/search?q=0x0dc411b17d337af85d83ea5a3577d09132aae866"
      >
        0x0dc411b17d337af85d83ea5a3577d09132aae866
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Date:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      Thursday, October 10th 2019, 06:13:48 AM
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Relayer:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy sc-bxivhb fqvWfz"
        href="/relayers/radar-relay"
      >
        <img
          class="sc-htpNat kedOmO sc-ifAKCX dAavUI"
          src="https://0xtracker.com/assets/logos/radar-relay.png"
        />
        Radar Relay
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Status:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <span
        class="sc-EHOje iekhCa badge badge-success"
      >
        successful
      </span>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      0x Protocol:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      v2
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Value:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <span>
        $184.29
      </span>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Maker Address:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/traders/0x2cfb4faaece38d5d03ef1ec9dfdf34964d6ab1b3"
      >
        0x2cfb4faaece38d5d03ef1ec9dfdf34964d6ab1b3
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Taker Address:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/traders/0x0dc411b17d337af85d83ea5a3577d09132aae866"
      >
        0x0dc411b17d337af85d83ea5a3577d09132aae866
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Maker Assets:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <ul
        class="list__List-sc-1hr42jb-0 cRmZyp"
      >
        <li
          class="list-item__Container-sc-1m8jbnt-0 kzXFZr"
        >
          <svg
            aria-hidden="true"
            class="sc-bdVaJa cjEDWs sc-bwzfXH hzBUAl"
            color="#9F9EB9"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 192 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
          <span>
            0.96955 
            <a
              class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
              href="/tokens/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
            >
              Wrapped Ether  (WETH)
            </a>
             
            <span
              class="asset-type-badge__AssetTypeBadge-sc-1ho1nac-0 ifWSro badge badge-dark"
            >
              ERC-20
            </span>
          </span>
        </li>
      </ul>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Taker Assets:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <ul
        class="list__List-sc-1hr42jb-0 cRmZyp"
      >
        <li
          class="list-item__Container-sc-1m8jbnt-0 kzXFZr"
        >
          <svg
            aria-hidden="true"
            class="sc-bdVaJa cjEDWs sc-bwzfXH hzBUAl"
            color="#9F9EB9"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 192 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
          <span>
            5,000 
            <a
              class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
              href="/tokens/0x4946fcea7c692606e8908002e55a582af44ac121"
            >
              FOAM Token  (FOAM)
            </a>
             
            <span
              class="asset-type-badge__AssetTypeBadge-sc-1ho1nac-0 ifWSro badge badge-dark"
            >
              ERC-20
            </span>
          </span>
        </li>
      </ul>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Derived Prices:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <ul
        class="list__List-sc-1hr42jb-0 cRmZyp"
      >
        <li
          class="list-item__Container-sc-1m8jbnt-0 kzXFZr"
        >
          <svg
            aria-hidden="true"
            class="sc-bdVaJa cjEDWs sc-bwzfXH hzBUAl"
            color="#9F9EB9"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 192 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
          <span>
            <a
              class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
              href="/tokens/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
            >
              Wrapped Ether  (WETH)
            </a>
            <span
              class="fill-details__PriceBadge-sc-5nb7jb-1 gmfFrK badge"
            >
              <span>
                $190.08
              </span>
            </span>
          </span>
        </li>
        <li
          class="list-item__Container-sc-1m8jbnt-0 kzXFZr"
        >
          <svg
            aria-hidden="true"
            class="sc-bdVaJa cjEDWs sc-bwzfXH hzBUAl"
            color="#9F9EB9"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 192 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
          <span>
            <a
              class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
              href="/tokens/0x4946fcea7c692606e8908002e55a582af44ac121"
            >
              FOAM Token  (FOAM)
            </a>
            <span
              class="fill-details__PriceBadge-sc-5nb7jb-1 gmfFrK badge"
            >
              <span>
                $0.0369
              </span>
            </span>
          </span>
        </li>
      </ul>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Maker Fee:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      None
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Taker Fee:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      None
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Fee Recipient:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/search?q=0xa258b39954cef5cb142fd567a46cddb31a670124"
      >
        0xa258b39954cef5cb142fd567a46cddb31a670124
      </a>
    </dd>
  </dl>
</DocumentFragment>
`;

exports[`fillDetails component should render V3 fill 1`] = `
<DocumentFragment>
  <dl
    class="fill-details__FillDetailList-sc-5nb7jb-0 mycsW"
  >
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Transaction Hash:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__OutboundLink-sc-1m1n17q-0 dFkevI"
        href="https://etherscan.io/tx/0x68e018d2c97cf76af6bae65297e6f5b6bc9680b95b828bdea5eb03a0b36ebc77"
        rel="noopener noreferrer"
        target="_blank"
      >
        0x68e018d2c97cf76af6bae65297e6f5b6bc9680b95b828bdea5eb03a0b36ebc77
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Order Hash:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/search?q=0x8104d4c4a32da7f1df357779fe6b8fc11dc6f4287de54566fe22eb35a74b0c71"
      >
        0x8104d4c4a32da7f1df357779fe6b8fc11dc6f4287de54566fe22eb35a74b0c71
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Sender Address:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/search?q=0x0dc411b17d337af85d83ea5a3577d09132aae866"
      >
        0x0dc411b17d337af85d83ea5a3577d09132aae866
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Date:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      Thursday, October 10th 2019, 06:13:48 AM
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Relayer:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy sc-bxivhb fqvWfz"
        href="/relayers/radar-relay"
      >
        <img
          class="sc-htpNat kedOmO sc-ifAKCX dAavUI"
          src="https://0xtracker.com/assets/logos/radar-relay.png"
        />
        Radar Relay
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Status:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <span
        class="sc-EHOje iekhCa badge badge-success"
      >
        successful
      </span>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      0x Protocol:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      v2
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Value:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <span>
        $184.29
      </span>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Maker Address:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/traders/0x2cfb4faaece38d5d03ef1ec9dfdf34964d6ab1b3"
      >
        0x2cfb4faaece38d5d03ef1ec9dfdf34964d6ab1b3
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Taker Address:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/traders/0x0dc411b17d337af85d83ea5a3577d09132aae866"
      >
        0x0dc411b17d337af85d83ea5a3577d09132aae866
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Maker Assets:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <ul
        class="list__List-sc-1hr42jb-0 cRmZyp"
      >
        <li
          class="list-item__Container-sc-1m8jbnt-0 kzXFZr"
        >
          <svg
            aria-hidden="true"
            class="sc-bdVaJa cjEDWs sc-bwzfXH hzBUAl"
            color="#9F9EB9"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 192 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
          <span>
            0.96955 
            <a
              class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
              href="/tokens/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
            >
              Wrapped Ether  (WETH)
            </a>
             
            <span
              class="asset-type-badge__AssetTypeBadge-sc-1ho1nac-0 ifWSro badge badge-dark"
            >
              ERC-20
            </span>
          </span>
        </li>
      </ul>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Taker Assets:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <ul
        class="list__List-sc-1hr42jb-0 cRmZyp"
      >
        <li
          class="list-item__Container-sc-1m8jbnt-0 kzXFZr"
        >
          <svg
            aria-hidden="true"
            class="sc-bdVaJa cjEDWs sc-bwzfXH hzBUAl"
            color="#9F9EB9"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 192 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
          <span>
            5,000 
            <a
              class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
              href="/tokens/0x4946fcea7c692606e8908002e55a582af44ac121"
            >
              FOAM Token  (FOAM)
            </a>
             
            <span
              class="asset-type-badge__AssetTypeBadge-sc-1ho1nac-0 ifWSro badge badge-dark"
            >
              ERC-20
            </span>
          </span>
        </li>
      </ul>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Derived Prices:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <ul
        class="list__List-sc-1hr42jb-0 cRmZyp"
      >
        <li
          class="list-item__Container-sc-1m8jbnt-0 kzXFZr"
        >
          <svg
            aria-hidden="true"
            class="sc-bdVaJa cjEDWs sc-bwzfXH hzBUAl"
            color="#9F9EB9"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 192 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
          <span>
            <a
              class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
              href="/tokens/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
            >
              Wrapped Ether  (WETH)
            </a>
            <span
              class="fill-details__PriceBadge-sc-5nb7jb-1 gmfFrK badge"
            >
              <span>
                $190.08
              </span>
            </span>
          </span>
        </li>
        <li
          class="list-item__Container-sc-1m8jbnt-0 kzXFZr"
        >
          <svg
            aria-hidden="true"
            class="sc-bdVaJa cjEDWs sc-bwzfXH hzBUAl"
            color="#9F9EB9"
            fill="currentColor"
            focusable="false"
            height="16"
            viewBox="0 0 192 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
          <span>
            <a
              class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
              href="/tokens/0x4946fcea7c692606e8908002e55a582af44ac121"
            >
              FOAM Token  (FOAM)
            </a>
            <span
              class="fill-details__PriceBadge-sc-5nb7jb-1 gmfFrK badge"
            >
              <span>
                $0.0369
              </span>
            </span>
          </span>
        </li>
      </ul>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Fee Recipient:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      <a
        class="link__InternalLink-sc-1m1n17q-1 kIJfjy"
        href="/search?q=0xa258b39954cef5cb142fd567a46cddb31a670124"
      >
        0xa258b39954cef5cb142fd567a46cddb31a670124
      </a>
    </dd>
    <dt
      class="fill-detail__Title-q3it9r-0 gEZdTC"
    >
      Protocol Fee:
    </dt>
    <dd
      class="fill-detail__Value-q3it9r-1 hhYdhc"
    >
      0.0001 ETH
      <span
        class="fill-details__PriceBadge-sc-5nb7jb-1 gmfFrK badge"
      >
        <span>
          $0.2000
        </span>
      </span>
    </dd>
  </dl>
</DocumentFragment>
`;
